Community/Tdarr_Plugin_MC93_Migz1FFMPEG.js

```diff
@@ -335,8 +364,72 @@ const plugin = (file, librarySettings, inputs, otherArguments) => {
     response.preset = '-c:v vp8_cuvid';
   }
 
-  response.preset += `${genpts}, -map 0 -c:v hevc_nvenc -cq:v 19 ${bitrateSettings} `
-  + `-spatial_aq:v 1 -rc-lookahead:v 32 -c:a copy -c:s copy -max_muxing_queue_size 9999 ${extraArguments}`;
+  // Determine if GPU is present and which gpu to use based on utilization of each GPU
+  response.infoLog += 'Running nvidia-smi to see if a GPU is available and selecting the GPU with less utilization\n';
+  let gpu_num = -1;
+  let gpu_util = 100000;
+  let result_util = 0;
+  let gpu_count = '';
+	let gpus_to_exclude = inputs.exclude_gpus.toLowerCase().split(",");
+  try {
+    let gpu_res = execSync('nvidia-smi --query-gpu=name --format=csv,noheader');
+    gpu_res = gpu_res.toString().trim();
+    gpu_res = gpu_res.split(/\r?\n/);
+    /* When nvidia-smi returns an error it contains 'nvidia-smi' in the error
+      Example: Linux: nvidia-smi: command not found
+               Windows: 'nvidia-smi' is not recognized as an internal or external command, 
+                   operable program or batch file. */
+    if (!gpu_res[0].includes('nvidia-smi')) {
+      gpu_count = gpu_res.length;
+    } else {
+      gpu_count = -1;
+    }
+  } catch (error) {
+    response.infoLog += 'Error in reading nvidia-smi output!\n';
+    response.infoLog += error.message;
+    gpu_count = -1;
+  }
+  if (gpu_count > 0) {
+    for (let gpui = 0; gpui < gpu_count; gpui++) {
+      // Check if GPU name is in GPUs to exclude
+      const is_gpu_excluded = gpus_to_exclude.filter(gpu_name => {
+        try {
+          if (gpu_count[i].toLowerCase().includes(gpu_name.toLowerCase())) {
+            return true;
+          }
```

Comment:
gpu_count isn't an array?

Reply:
that was an error, changed it to `gpu_res[gpui]`